src/gatsby/models/authors/onCreateMarkdownContentNode.js

```diff
@@ -1,16 +1,50 @@
+function parseLink(input) {
+  if (typeof input === 'string') {
```

Comment:
why do we need this check?

Reply:
I mentioned in the original spec that we should allow for both strings or objects when defining Author links. Both methods handle different situations well, and allowing for both not only makes up for the weaknesses of each alone but is also very easy and natural to YAML Markdown frontmatter.

Strings are simple, easy to use and pack a lot of info for known sites like Twitter. Supporting custom objects, however, allows for total control in special cases like having a personal blog, a portfolio site, or simply wanting the Twitter link to have a description associated with it if we source this data in a more detailed context.

This support would be particularly useful for the new Author pages at #1364.

I agree that the function isn't very expressively written though, I'll try to improve that.

Reply:
The new parser makes it more obvious that it's handling both objects and strings.